src/averageAges.js

```diff
@@ -55,7 +84,39 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const onlyWomen = people.filter(person => person.sex === 'f');
+
+  const womenWithChildren = onlyWomen.filter(women => {
+    return people.some(person => person.mother === women.name);
+  });
+
+  let childrenCount = 0;
+
+  const totalAgeDiff = womenWithChildren.reduce((acc, women) => {
+    const ageDiff = ageDiffOfChildren(women);
+
+    return acc + ageDiff;
+  }, 0);
+
+  function ageDiffOfChildren(women) {
```

Comment:
try to get rid of this function, you use it only once, so it doesn't worth to move the code into separated function